lib/index.js

```diff
@@ -1723,38 +1749,35 @@ exposeMethod('pluck');
  */
 
 Stream.prototype.pickBy = function (f) {
-
-    return this.consume(function (err, x, push, next) {
+    return this.map(objectOnly(function (x) {
         var out = {};
-        if (err) {
-            push(err);
-            next();
+        var seen = {};  // prevents testing overridden properties multiple times.
```

Comment:
The best way to deal with it is to drop support for IE8. ;) Yeah, I guess we could do

```javascript
var seen = isES5 ? Object.create(null) : {}
```

And then the strict equals test (that's very nice, I never knew about that before).